core/src/org/sbml/jsbml/validator/SyntaxChecker.java

```diff
@@ -204,31 +214,61 @@ public static final boolean isValidId(String idCandidate, int level,
    *         {@link String}, {@code false} otherwise.
    */
   public static final boolean isValidMetaId(String idCandidate) {
-    if ((syntaxChecker.simpleMetaIdPattern == null) || (syntaxChecker.metaIdPattern == null)) {
-      syntaxChecker.initMetaIdPatterns();
-    }
+    return isValidMetaId(idCandidate, 2, 2); //TODO - should version 2 or 4 be the default
+  }
+  
+  /**
+   * Checks if the given identifier candidate satisfies the requirements for a
+   * valid meta identifier (for any version and level).
+   *
+   * @param idCandidate, level, version
+   * @return {@code true} if the given argument is a valid meta identifier
+   *         {@link String}, {@code false} otherwise.
+   */
+  public static final boolean isValidMetaId(String idCandidate, int level, int version) {
 
-    // In the most cases the first check will be sufficient.
-    if (syntaxChecker.simpleMetaIdPattern.matcher(idCandidate).matches()) {
-      return true;
-    }
-    /* In the worst case, we have to perform two checks, but this is
-     * hopefully much faster than doing the full check each time.
-     * It can be assumed that the majority of allowable characters is
-     * not used within these identifiers.
+    String metaId = syntaxChecker.initMetaIdPatterns(level, version);
```

Comment:
@Onur101010  calling this string metaId is quite misleading. It is always better to try to give a name describing the variable properly. In this case, I would call it something like metaIdPatternKey.